src/CometFlexVoting.sol

```diff
@@ -0,0 +1,45 @@
+// SPDX-License-Identifier: UNLICENSED
+pragma solidity >=0.8.10;
+
+import {Comet} from "comet/Comet.sol";
+import {CometConfiguration} from "comet/CometConfiguration.sol";
+import {Checkpoints} from "@openzeppelin/contracts/utils/Checkpoints.sol";
+
+import {FlexVotingClient} from "src/FlexVotingClient.sol";
+
+// TODO add description
+contract CometFlexVoting is Comet, FlexVotingClient {
+  using Checkpoints for Checkpoints.History;
+
+  /// @param _config The configuration struct for this Comet instance.
+  /// @param _governor The address of the flex-voting-compatible governance contract.
+  constructor(CometConfiguration.Configuration memory _config, address _governor)
+    Comet(_config)
+    FlexVotingClient(_governor)
+  {
+    _selfDelegate();
```

Comment:
Hm, good question.

We aren't currently doing it in the constructor because if the client/pool is a proxy then it has to self delegate some other way (e.g. within an initializer). But I don't necessarily see a drawback to self delegating in both places. If a logic contract has self-delegated (because self-delegation happens in its constructor) who cares -- it won't even hold any tokens so it shouldn't matter

cc @mds1 in case there's something I'm missing here

Reply:
Just for reference the method looks like this:

```solidity
/// @dev Delegates the present contract's voting rights with `GOVERNOR` to itself.
function _selfDelegate() internal {
  IVotingToken(GOVERNOR.token()).delegate(address(this));
}
```

And we set the `GOVERNOR` in the constructor of `FlexVotingClient`:

```solidity
/// @param _governor The address of the flex-voting-compatible governance contract.
constructor(address _governor) {
  GOVERNOR = IFractionalGovernor(_governor);
}
```

 So IF we called `_selfDelegate` in the constructor of `FlexVotingClient` AND someone wanted to deploy a logic (implementation) contract for a proxy that used `FlexVotingClient`, THEN they'd have to set a real governor in their logic contract, otherwise the constructor would revert because the call would fail. Personally I like using dummy/null values to be set in logic contracts, as it makes their purpose more clear.

It seems we could either:
1. Leave things as-is, this gives implementers the most optionality
2. Update `_selfDelegate` to be a no-op if the governor address is the zero address (limits what the "dummy value" for the governor can be in the logic contract)
3. Update `_selfDelegate` to have a try/catch so we can call it in the constructor and let people use any dummy values in their logic contracts

Option 1 seems the simplest to me.

But to this point, aren't we currently inconsistent with regard to proxies? You can only set `GOVERNOR` in the constructor, so we don't support proxies nicely anyway, i.e. you can't have a proxy factory that spits out flex voting clients with different governors. So maybe the `FlexVotingClient` needs to replace the constructor with an `initialize` method that both sets the governor and then self-delegates.

Reply:
Great point about proxies -- added this motivation to our existing ticket to make `GOVERNOR` mutable, https://github.com/ScopeLift/flexible-voting/issues/48

I agree that we should just leave as-is for now

Reply:
Hmm yeah all good points about the proxies. On the one hand I think that deploying Governors via Proxy isn't currently something I see happening, and I'm not sure it makes a lot of sense. That said, I don't think we should foreclose on working well with them. So we should consider it more carefully. Let's pull out an issue to consider Proxy support and link to this convo. Otherwise can be closed.

Reply:
> On the one hand I think that deploying Governors via Proxy isn't currently something I see happening

Just to clarify, this is about deploying the _voting client_, not the governor, and clients are often proxies (e.g. AToken contracts)

Reply:
Yep good point. I was mixed up when I wrote that, thanks. We should definitely support Proxies for clients.